UnitTests/GitUITests/CommandsDialogs/CommitDialog/FormCommitTests.cs

```diff
@@ -88,6 +88,8 @@ public void Should_display_branch_and_no_remote_info_in_statusbar()
             _referenceRepository.CheckoutMaster();
             RunFormTest(form =>
             {
+                form.GetTestAccessor().RunningTasks.Join();
```

Comment:
💭 Not sure if we need to use this or `JoinTillEmptyAsync`

/cc @AArnott

Reply:
Oh boy, I forgot the Join method even existed on that collection type. That's an unfortunate name, since `JoinableTask.Join()` is a blocking call, one might expect *this* to be as well. But it isn't. This immediately returns. You need to call `JoinTillEmptyAsync` (and await or synchronously block with JTF.Run) in order to suspend your test till the operations are done.

---
In reply to: [258626551](https://github.com/gitextensions/gitextensions/pull/6273#discussion_r258626551) [](ancestors = 258626551)

Reply:
> ...one might expect _this_ to be as well. But it isn't. This immediately returns.

Aha! That probably explains why the test continued to fail.

@sharwell I'm a bit at loss how to correctly await `JoinTillEmptyAsync` call in the test:
```
            RunFormTest(form =>
            {
                form.GetTestAccessor().RunningTasks.JoinTillEmptyAsync();

                var currentBranchNameLabelStatus = form.GetTestAccessor().CurrentBranchNameLabelStatus;
                var remoteNameLabelStatus = form.GetTestAccessor().RemoteNameLabelStatus;

                Assert.AreEqual("master →", currentBranchNameLabelStatus.Text);
                Assert.AreEqual("(remote not configured)", remoteNameLabelStatus.Text);
            });
```

How would you await it?

Reply:
If you can adjust `RunFormTest` to take a `Func<Task>` instead of (or in addition to) an `Action`, then  you can make this entire delegate async and then just *await*. That would be ideal since it lets your test emulate a real user who isn't always controlling the UI thread.
But your other option is to simply call:
```cs
joinableTaskFactory.Run(() => form.GetTestAccessor().RunningTasks.JoinTillEmptyAsync());
```